src/main/kotlin/com/terraformation/backend/accelerator/db/ParticipantProjectSpeciesStore.kt

```diff
@@ -119,6 +132,79 @@ class ParticipantProjectSpeciesStore(
         .execute()
   }
 
+  fun fetchParticipantProjectsForSpecies(
+      organizationId: OrganizationId,
+      speciesId: SpeciesId
+  ): List<ParticipantProjectsForSpecies> {
+    val today = LocalDate.ofInstant(clock.instant(), TimeZones.UTC)
+    val user = currentUser()
+
+    return dslContext
+        .select(
+            DELIVERABLES.ID,
+            PROJECTS.NAME,
+            PROJECTS.ID,
+            PARTICIPANT_PROJECT_SPECIES.ID,
+            PARTICIPANT_PROJECT_SPECIES.SUBMISSION_STATUS_ID,
+            SPECIES.ID)
+        .from(SPECIES)
+        .join(PARTICIPANT_PROJECT_SPECIES)
+        .on(SPECIES.ID.eq(PARTICIPANT_PROJECT_SPECIES.SPECIES_ID))
+        .join(PROJECTS)
+        .on(PARTICIPANT_PROJECT_SPECIES.PROJECT_ID.eq(PROJECTS.ID))
+        .join(ORGANIZATIONS)
+        .on(PROJECTS.ORGANIZATION_ID.eq(ORGANIZATIONS.ID))
+        .join(PARTICIPANTS)
+        .on(PROJECTS.PARTICIPANT_ID.eq(PARTICIPANTS.ID))
+        .leftOuterJoin(COHORT_MODULES)
+        .on(
+            COHORT_MODULES.COHORT_ID.eq(PARTICIPANTS.COHORT_ID),
+            COHORT_MODULES.END_DATE.greaterOrEqual(today),
+            COHORT_MODULES.START_DATE.lessOrEqual(today))
+        .leftOuterJoin(MODULES)
+        .on(COHORT_MODULES.MODULE_ID.eq(MODULES.ID))
+        .leftOuterJoin(DELIVERABLES)
+        .on(
+            DELIVERABLES.MODULE_ID.eq(MODULES.ID),
+            DELIVERABLES.DELIVERABLE_TYPE_ID.eq(DeliverableType.Species))
+        .where(ORGANIZATIONS.ID.eq(organizationId))
+        .and(SPECIES.ID.eq(speciesId))
+        .fetch {
+          if (user.canReadProjectDeliverables(it[PROJECTS.ID]!!)) {
+            ParticipantProjectsForSpecies.of(it)
+          } else {
+            it[DELIVERABLES.ID] = null
+            ParticipantProjectsForSpecies.of(it)
+          }
+        }
+        .filter { user.canReadProject(it.projectId) }
+  }
+
+  fun fetchSpeciesForParticipantProjects(projectId: ProjectId): List<SpeciesForParticipantProject> {
+    val user = currentUser()
+
+    return dslContext
+        .select(
+            PROJECTS.NAME,
+            PROJECTS.ID,
+            PARTICIPANT_PROJECT_SPECIES.ID,
+            PARTICIPANT_PROJECT_SPECIES.RATIONALE,
+            PARTICIPANT_PROJECT_SPECIES.SUBMISSION_STATUS_ID,
+            SPECIES.ID,
+            SPECIES.COMMON_NAME,
+            SPECIES.SCIENTIFIC_NAME)
+        .from(SPECIES)
+        .join(PARTICIPANT_PROJECT_SPECIES)
+        .on(SPECIES.ID.eq(PARTICIPANT_PROJECT_SPECIES.SPECIES_ID))
+        .join(PROJECTS)
+        .on(PARTICIPANT_PROJECT_SPECIES.PROJECT_ID.eq(PROJECTS.ID))
+        .join(PARTICIPANTS)
+        .on(PROJECTS.PARTICIPANT_ID.eq(PARTICIPANTS.ID))
```

Comment:
Is this just a guard against cases where there's a participant project species for a project that isn't in a participant? If so, it'd also work (and would remove an unnecessary join) to do `.and(PROJECTS.PARTICIPANT_ID.isNotNull)` in the "where" clause.

Reply:
Yes, that is the purpose. It _should_ be impossible for it to exist on a project that isn't associated to a participant, so perhaps neither join or condition are necessary.